worker/src/main/java/gov/cms/ab2d/worker/processor/TextStreamHelperImpl.java

```diff
@@ -0,0 +1,93 @@
+package gov.cms.ab2d.worker.processor;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.*;
+import java.nio.file.Path;
+
+/**
+ * Implement a plain text stream helper
+ */
+@Slf4j
+public class TextStreamHelperImpl extends StreamHelperImpl {
+
+    /**
+     * Implement the text stream helper
+     *
+     * @param path - where to create the file
+     * @param contractNumber - the contract number
+     * @param totalBytesAllowed - the total number of bytes allowed to be written to the stream
+     * @throws FileNotFoundException - if the file can't be created
+     */
+    TextStreamHelperImpl(Path path, String contractNumber, long totalBytesAllowed, int tryLockTimeout)
+            throws FileNotFoundException {
+        super(path, contractNumber, totalBytesAllowed, tryLockTimeout);
+
+        setCurrentStream(createStream());
+    }
+
+    /**
+     * Create the next stream from the next file name
+     *
+     * @return the stream
+     * @throws FileNotFoundException if you can't create the stream
+     */
+    private OutputStream createStream() throws FileNotFoundException {
+        String fileName = getPath().toString() + "/" + createFileName();
+        File f = new File(fileName);
+        f.getParentFile().mkdirs();
+        OutputStream stream = new BufferedOutputStream(new FileOutputStream(fileName));
+        Path p = Path.of(fileName);
+        getFilesCreated().add(p);
+        return stream;
+    }
+
+    /**
+     * Write data to the correct file and iterate to the next file if it exceeds the limit
+     *
+     * @param data - the data to write
+     */
+    @Override
+    public void addData(byte[] data) {
+        tryLock(getDataFileLock());
+        if (data == null || data.length == 0) {
+            return;
+        }
+        try {
+            if (getCurrentStream() == null) {
+                setCurrentStream(createStream());
+                setTotalBytesWritten(0);
+            }
+            if (getTotalBytesWritten() + data.length > getTotalBytesAllowed() && getTotalBytesWritten() > 0) {
+                getCurrentStream().close();
+                setCurrentStream(createStream());
+                setTotalBytesWritten(0);
+            }
+            getCurrentStream().write(data);
+            setTotalBytesWritten(getTotalBytesWritten() + data.length);
+        } catch (Exception ex) {
+            log.error("Unable to create file output stream for contract " + getContractNumber() + "[" + (getCounter() - 1) + "]", ex);
+        } finally {
+            getDataFileLock().unlock();
+        }
+    }
+
+    /**
+     * Close the stream clean up any empty files in the files created list
+     */
+    @Override
+    public void close() {
+        if (getCurrentStream() == null) {
+            return;
+        }
+        try {
+            getCurrentStream().close();
+            int numFiles = getFilesCreated().size();
+            if (getFilesCreated().get(numFiles - 1).toFile().length() == 0) {
```

Comment:
We need a index check here. In case BFD returns _no_ patients for a contract, list of files will be empty.

Reply:
...need a test to cover this scenario as well

Reply:
Since we add the file to the list when we create the stream, the testing of the length of the last file is doing that. If no data exists, it's a list of one that has no data in it, but this also allows us to check for instances where the boundary between files occurs exactly at the end of the data to add.